### PR TITLE
[xray] Optional support for Postgres with TLS

### PR DESCRIPTION
**Pull request by rimusz**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Chart Version bumped
- [X] CHANGELOG.md updated
- [X] Variables and other changes are documented in the README.md

### CODE REVIEW

File: stable/xray/values.yaml [hunk 1]
```diff
@@ -267,7 +267,9 @@ common:
 # For setting up external services, must pass the connection URL for them
 global:
   mongoUrl:
-  postgresqlUrl:
+  postgresql:
```
Comment (eldada) [inline stable/xray/values.yaml]: For not breaking compatibility, need to also support old `postgresqlUrl`. Or existing users' deployments will break.

File: stable/xray/templates/xray-setup-conf.yaml [hunk 2]
```diff
@@ -15,11 +15,20 @@ data:
...
     XRAYCONFIGPATH={{ .Values.common.xrayConfigPath }}
     XRAY_CONFIG_DIR=${XRAYCONFIGPATH}/config
+    XRAY_TLS_DIR=${XRAYCONFIGPATH}/tls
     XRAY_CONFIG_FILE=${XRAY_CONFIG_DIR}/xray_config.yaml
 
     echo "Creating directories"
     mkdir -pv ${XRAY_CONFIG_DIR}
 
+    # Set Postgres TLS certs
+    {{- if and .Values.global.postgresql.url .Values.global.postgresql.tlsSecret (not .Values.postgresql.enabled) }}
```
Comment (eldada) [inline stable/xray/templates/xray-setup-conf.yaml]: Let's add a check that the `postgresql.url` contains the `sslkey=` and `sslrootcert=`, so we validate users actually put this path of the url and didn't forget. WDYT?

File: stable/xray/templates/xray-setup-conf.yaml [hunk 3]
```diff
@@ -15,11 +15,20 @@ data:
     SCRIPTS_DIR=/scripts
     XRAYCONFIGPATH={{ .Values.common.xrayConfigPath }}
     XRAY_CONFIG_DIR=${XRAYCONFIGPATH}/config
+    XRAY_TLS_DIR=${XRAYCONFIGPATH}/tls
     XRAY_CONFIG_FILE=${XRAY_CONFIG_DIR}/xray_config.yaml
 
     echo "Creating directories"
     mkdir -pv ${XRAY_CONFIG_DIR}
 
+    # Set Postgres TLS certs
```
[the remark this answers is not in the document]
Comment (rimusz) [reply stable/xray/templates/xray-setup-conf.yaml]: Ss setting `defaultMode` is not fully respected when `Pod.spec.securityContext.runAsUser` is set, `client.key` needs permissions `0600` or less and mounting it directly from secret Xray fails.
https://github.com/kubernetes/kubernetes/issues/57923

File: stable/xray/README.md [hunk 4]
```diff
@@ -162,10 +164,39 @@ For this, pass the parameters: `postgresql.enabled=false` and `global.postgresql
...
+# Example
+# PostgreSQL host: custom-postgresql.local
+# PostgreSQL port: 5432
+# PostgreSQL user: xray
+# PostgreSQL password: password2_X
+# PostgreSQL server CA: server-ca.pem
+# PostgreSQL client cert: client-key.pem
+# PostgreSQL clinet key: client-cert.pem
+
+export XRAY_POSTGRESQL_CONN_URL="postgres://${POSTGRESQL_USER}:${POSTGRESQL_PASSWORD}@custom-postgresql.local:5432/${POSTGRESQL_DATABASE}?sslrootcert=/var/opt/jfrog/xray/data/tls/${POSTGRESQL_SERVER_CA}&sslkey=/var/opt/jfrog/xray/data/tls/${POSTGRESQL_CLIENT_KEY}&sslcert=/var/opt/jfrog/xray/data/tls/${POSTGRESQL_CLIENT_CERT}&sslmode=verify-ca"
```
[the remark this answers is not in the document]
Comment (rimusz) [reply stable/xray/README.md]: as consuming the whole URL makes easier and it falls back to the same setup way as we have for non TLS setup

[the remark this answers is not in the document]
Comment (rimusz) [reply stable/xray/README.md]: we already have very complicated Artifactory chart, I do not want to make the Xray one too complicated as well :)

[the remark this answers is not in the document]
Comment (danielezer) [reply stable/xray/README.md]: fair enough :)

File: stable/xray/templates/xray-setup-conf.yaml (hunk 2 above)
Comment (rimusz) [reply stable/xray/templates/xray-setup-conf.yaml]: not sure how to make not very complicated way, for now I improved examples how to set it up.

**Pull request merged**